### PR TITLE
Debug gesture and logout OAuth endpoint

### PR DESCRIPTION
**Pull request by adamhrz**
Adds a two finger triple tap gesture to show the debug menu: 
* Shows build number
* Simulates OAuth bad refresh token
* Simulates OAuth bad access token
* Can be used to force a logout

### CODE REVIEW

File: PRODUCTNAME/app/PRODUCTNAME/Application/AppDelegate.swift [hunk 1]
```diff
@@ -26,7 +26,83 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
             config.onDidLaunch(application: application, launchOptions: launchOptions)
         }
 
+        // Don't load the main UI if we're unit testing.
+        if let _: AnyClass = NSClassFromString("XCTest") {
         return true
     }
```
Comment (KingOfBrian) [inline PRODUCTNAME/app/PRODUCTNAME/Application/AppDelegate.swift]: I think the indentation is off here, and I'd prefer personally to run this before the configurations ran. Often tests want to run without the app configurations running.

File: PRODUCTNAME/app/PRODUCTNAME/Application/AppDelegate.swift [hunk 2]
```diff
@@ -26,7 +26,83 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
 
+        window = UIWindow(frame: UIScreen.main.bounds)
+
+        // Adds (by default) a 2 finger triple tap gesture to present a debug menu
+        enableDebugGesture()
```
Comment (KingOfBrian) [inline PRODUCTNAME/app/PRODUCTNAME/Application/AppDelegate.swift]: Actually, can we move this entire behavior into an `AppLifecycle `? It'd be encapsulated nicely and thin out the AppDelegate.

File: PRODUCTNAME/app/PRODUCTNAME/Application/AppDelegate.swift [hunk 3]
```diff
@@ -26,7 +26,83 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+
+        if let w = window {
+            // Configure global window defaults
+            w.backgroundColor = UIColor.white
+
+            // Setting a tintcolor on the window adds a tint to all UIAlertControllers
+            w.tintColor = UIColor.blue
```
Comment (KingOfBrian) [inline PRODUCTNAME/app/PRODUCTNAME/Application/AppDelegate.swift]: I'd prefer to leave styling out for the template, but not a big deal

File: PRODUCTNAME/app/PRODUCTNAME/Application/AppDelegate.swift (hunk 2 above)
Comment (adamhrz) [inline PRODUCTNAME/app/PRODUCTNAME/Application/AppDelegate.swift]: 👍

File: PRODUCTNAME/app/PRODUCTNAME/Application/AppDelegate.swift (hunk 3 above)
Comment (adamhrz) [inline PRODUCTNAME/app/PRODUCTNAME/Application/AppDelegate.swift]: My thought was something to suggest where app hide window default would happen - happy to take it out

**Comment by adamhrz** [comment]
@KingOfBrian This is ready for another look. I addressed your concerns moving the debug menu a config, removed the placeholder empty VC and re-generated the cookie cutter template

File: PRODUCTNAME/app/PRODUCTNAME/Application/AppDelegate.swift [hunk 4]
```diff
@@ -18,15 +18,36 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
 
+        // Don't load the main UI if we're unit testing.
+        if let _: AnyClass = NSClassFromString("XCTest") {
+            return true
+        }
+
+        window = UIWindow(frame: UIScreen.main.bounds)
+        showTabBar(animated: false)
+        window?.makeKeyAndVisible()
```
Comment (KingOfBrian) [inline PRODUCTNAME/app/PRODUCTNAME/Application/AppDelegate.swift]: Can we move `showTabBar` and the `makeKeyAndVisible` to after the config?

Comment (KingOfBrian) [inline PRODUCTNAME/app/PRODUCTNAME/Application/AppDelegate.swift]: Also, maybe make `showTabBar` more generic, like `configureRootViewController`?

**Pull request merged**